locator/tablets.cc

```diff
@@ -1101,6 +1101,86 @@ void assert_rf_rack_valid_keyspace(std::string_view ks, const token_metadata_ptr
     tablet_logger.debug("[assert_rf_rack_valid_keyspace]: Keyspace '{}' has been verified to be RF-rack-valid", ks);
 }
 
+void validate_rf_rack_valid_replication(const token_metadata_ptr tmptr, const replica::keyspace& ks) {
+    auto ks_name = ks.metadata()->name();
+    tablet_logger.debug("[validate_rf_rack_valid_replication]: Starting verifying that all tables in keyspace '{}' are RF-rack-valid", ks_name);
+    const auto& ars = ks.get_replication_strategy();
+    // Any keyspace that does NOT use tablets is RF-rack-valid.
+    if (!ars.uses_tablets()) {
+        tablet_logger.debug("[validate_rf_rack_valid_replication]: Keyspace '{}' has been verified to be RF-rack-valid (no tablets)", ks_name);
+        return;
+    }
+
+    // Tablets can only be used with NetworkTopologyStrategy.
+    SCYLLA_ASSERT(ars.get_type() == replication_strategy_type::network_topology);
+    const auto& nts = *static_cast<const network_topology_strategy*>(std::addressof(ars));
+
+    std::map<sstring, int> rf_per_dc;
+    bool all_dc_rf_lower_than_2 = true;
```

Comment:
What's special about 2?

Reply:
In the patch introducing the rf-rack-valid feature we decided (I'm not sure for which reasons) that keyspaces with RF=1 should be treated as rf-rack-valid, even though their tablets could migrate between racks. RF=0 should also be rf-rack-valid, so a bool for both of them is RF<2

Reply:
I remember we wanted to treat #racks=1 as special, maybe RF=1 got confused with it.

Reply:
@bhalevy suggests that it's because there are no pairing problems with RF=1. But I'd rather eliminate special cases for unimportant workloads.